src/drive/lib/encryption.js

```diff
@@ -1,3 +1,80 @@
+const stringToArrayBuffer = string => {
+  var encoder = new TextEncoder('utf-8')
+  return encoder.encode(string)
+}
+
+const asDerivableKey = async (
+  baseMaterial,
+  keyDerivationAlgorithm = 'PBKDF2'
+) => {
+  let usableMaterial
+  if (baseMaterial instanceof CryptoKey) {
+    usableMaterial = await window.crypto.subtle.exportKey('raw', baseMaterial)
+  } else {
+    usableMaterial = stringToArrayBuffer(baseMaterial)
+  }
+  return await window.crypto.subtle.importKey(
+    'raw',
+    usableMaterial,
+    { name: keyDerivationAlgorithm },
+    true,
+    ['deriveKey']
+  )
+}
+
+const slowHashing = async (
+  baseKey,
+  saltBuffer,
+  { keyDerivationAlgorithm, iterations, hash } = {},
+  { algorithm, keyLength } = {}
+) => {
+  return await window.crypto.subtle.deriveKey(
+    {
+      name: keyDerivationAlgorithm || 'PBKDF2',
+      salt: saltBuffer,
+      iterations: iterations || 1000,
+      hash: hash || 'SHA-256'
+    },
+    baseKey,
+    {
+      name: algorithm || 'AES-GCM',
+      length: keyLength || 256
+    },
+    true,
+    ['encrypt', 'decrypt', 'wrapKey', 'unwrapKey']
+  )
+}
+
+export const deriveKey = async (password, salt) => {
+  const passwordAsKey = await asDerivableKey(password)
+  const saltBuffer = stringToArrayBuffer(salt)
+  const passwordBuffer = stringToArrayBuffer(password)
+  // Chain 2 key derivations : first, derive a key from a password
+  const preKey = await slowHashing(passwordAsKey, saltBuffer, {
+    iterations: 1000
+  })
+  const key = await asDerivableKey(preKey).then(key => {
+    return slowHashing(key, passwordBuffer, { iterations: 1 })
+  })
+  return key
+}
+
+export const exportKey = async (key, vaultKey) => {
+  return window.crypto.subtle.wrapKey('jwk', key, vaultKey, { name: 'AES-KW' })
```

Comment:
Reading the docs, it seems it should be an object `AesCbcParams` : https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/wrapKey#Parameters

Reply:
Wrap algo could be any AES-* algorithm. AES-KW is also accepted :
"To use AES-KW, pass the string "AES-KW", or an object of the form { "name": "AES-KW }."

Reply:
Ok, I thought it was necessary to keep the same algo than the one used for data encryption. 
I'm a bit confused by the fact that others AES methods are available if `AES-KW` is dedicated for key wraping, though